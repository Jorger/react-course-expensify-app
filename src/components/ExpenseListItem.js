import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({id, description, amount, createAt}) => (
    <div>
        <h3>
        <Link to={`/edit/${id}`}>
            {description}
        </Link>
        </h3>
        <p>
            {numeral(amount / 100).format('$0,0.00')}
            -
            {moment(createAt).format('Do MMMM, YYYY')}
        </p>
    </div>
);

//export default connect()(ExpenseListItem);

export default ExpenseListItem;