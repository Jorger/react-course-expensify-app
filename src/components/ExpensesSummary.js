import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => (
    <div>
        {
            <h1>
                Viewing {expensesCount} {expensesCount >= 2 ? "Expenses" : "Expense"} totalling {numeral(expensesTotal / 100).format('$0,0.00')}
            </h1>
        }
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(
        state.expenses, state.filters
    );
    return {
        expensesCount : visibleExpenses.length, 
        expensesTotal : selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);