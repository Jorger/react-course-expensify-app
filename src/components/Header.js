import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Espensify</h1>
        <NavLink to="/" activeClassName="is-active" exact>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create expense</NavLink>
    </header>
);

export default Header;