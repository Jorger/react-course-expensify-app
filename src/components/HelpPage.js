import React from 'react';

const HelpPage = () => (
    <div>
        PÃ¡gina de <strong>Help Page</strong> component
    </div>
);

export default HelpPage;