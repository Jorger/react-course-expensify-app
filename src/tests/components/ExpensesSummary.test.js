import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import selectExpensesTotal from '../../selectors/expenses-total';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary correctly', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={100}/>);
    expect(wrapper).toMatchSnapshot();
});


test('should render ExpensesSummary with multiple expenses', () => {
    const expensesCount = expenses.length;
    const expensesTotal = selectExpensesTotal(expenses);
    const wrapper = shallow(<ExpensesSummary expensesCount={expensesCount} expensesTotal={expensesTotal} />);
    expect(wrapper).toMatchSnapshot();
});