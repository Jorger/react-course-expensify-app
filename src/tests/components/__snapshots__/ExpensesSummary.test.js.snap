// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    Expense
     totalling 
    $1.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    Expenses
     totalling 
    $1,141.95
  </h1>
</div>
`;
